File: practices/backends/go-gin/routes/heartbeat.go
package routes

import (
	"gin-app/types"
	"net/http"
	"time"

	"github.com/gin-gonic/gin"
)

func HeartbeatRouter(c *gin.Context) {
	now := time.Now()
	response := types.HeartbeatResponse{
		Alive: now.Format(time.RFC3339),
	}
	c.JSON(http.StatusOK, response)
}

File: practices/backends/typescript-express/src/routes/heartbeat.ts
import type { Request, Response } from "express";
import type { HeartbeatResponse } from "src/types/heartbeat";

const heartbeat = async (req: Request, res: Response) => {
  console.log("req:", req.body);
  const now = new Date();
  const response: HeartbeatResponse = {
    alive: now.toISOString(),
  };
  res.json(response);
};

export default heartbeat;
